File: client/app/scripts/services/invoiceService.js
/**
 * Created by Sean on 6/18/2014.
 */
angular.module('newlpApp')
    .factory('Invoice', function ($http, $resource) {
        return $resource('/api/v1/invoices/:invoiceId', {invoiceId: "@invoiceId"}, {
            update: {
                method: 'PUT'
            },
            patch: {
                method: 'PATCH'
            },
//            save: {
//                method: 'POST',
//                params:{},
//                isArray:false,
//                headers:{
//                    'Content-Type':'application/json'
//                }
//            },
            getAll: {
                method: 'GET',
                params: {
                }
            },
            queryByAuditStatus: {
                url: '/api/v1/invoices/search/auditStatus/:auditStatus',
                method: 'GET',
                params: {
                    auditStatus: '@auditStatus'
                }
            },
            queryByOriginalAuditStatus: {
                url: '/api/v1/invoices/search/audited/auditStatus/:auditedStatus',
                method: 'GET',
                params: {
                    auditedStatus: '@auditedStatus'
                }
            },
            search: {
                url: '/api/v1/invoices/search',
                method: 'POST',
                params: {
                },
                isArray: false,
                headers: {
                    'Content-Type': 'application/json'
                }
            },
            searchByOriginalAuditStatus: {
                url: '/api/v1/invoices/search',
                method: 'POST',
                params: {
                },
                isArray: false,
                headers: {
                    'Content-Type': 'application/json'
                }
            },
            getPreAuditStatusByInvoiceId:{
                url: '/api/v1/invoices/:invoiceId/auditStatus/getPreAuditStatusByInvoiceId',
                method: 'GET',
                params: {
                    invoiceId: '@invoiceId'
                },
                isArray: false,
                headers: {
                    'Content-Type': 'application/json'
                }
            },
            getSendBackAuditStatus:{
                url: '/api/v1/invoices/:invoiceId/auditStatus/getSendBackAuditStatus',
                method: 'GET',
                params: {
                    invoiceId: '@invoiceId'
                },
                isArray: false,
                headers: {
                    'Content-Type': 'application/json'
                }
            }
        })
    })
;

File: server/src/main/java/com/lesso/newlp/invoice/service/InvoiceService.java
package com.lesso.newlp.invoice.service;

import com.lesso.newlp.invoice.entity.InvoiceEntity;
import com.lesso.newlp.invoice.model.SearchTerm;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.lang.reflect.InvocationTargetException;
import java.util.Set;

/**
 * Created by Sean on 6/20/2014.
 */
public interface InvoiceService {

    InvoiceEntity save(InvoiceEntity invoice);

    InvoiceEntity findById(Long invoiceId) throws Exception;

    InvoiceEntity update(InvoiceEntity invoice) throws InvocationTargetException, IllegalAccessException;

    InvoiceEntity patch(Long invoiceId, InvoiceEntity invoice) throws InvocationTargetException, IllegalAccessException;

    void delete(Long invoiceId);

    Page<InvoiceEntity> queryByAuditStatus(Integer auditStatus,String memberId, Pageable pageable);

    Page<InvoiceEntity> search(SearchTerm searchTerm, Pageable pageable,String memberId);

    Integer getPreAuditStatusByInvoiceId(Long invoiceId);

    int queryCountByAuditStatus(Set<Integer> auditStatuses, String username);

    int queryCountByOriginalAuditStatus(int auditedStatus, String username);

    Page<InvoiceEntity> searchByOriginalAuditStatuses(Integer auditedStatus, String username, Pageable pageable);

    Page<InvoiceEntity> searchByOriginalAuditStatus(SearchTerm searchTerm, Integer auditedStatus, Pageable pageable, String username);

    Integer getSendBackAuditStatus(Long invoiceId);
}
